### PR TITLE
Updates webots setup and running webots sections with `./b` stuff

### PR DESCRIPTION
**Pull request by KipHamiltons**
I added all of the `configure` flags from the cmake file. lmk what you think

### CODE REVIEW

File: src/book/03-guides/02-tools/03-webots.mdx [hunk 1]
```diff
@@ -96,25 +96,27 @@ sudo apt-get install cmake-curses-gui libprotobuf-dev protobuf-compiler libeigen
...
-2. Compile the controllers with cmake
+2. Configure the codebase by running
 
-   1. Make a build folder and move into it
-
-      ```sh
-      mkdir build && cd build
-      ```
+   ```sh
+   ./b configure [-i] [-- <flags>]
```
Comment (ysims) [inline src/book/03-guides/02-tools/03-webots.mdx:102]: This might be giving people too many choices in a starting guide - it's still good to have, but maybe make a note that if you're not sure what to run just run `./b configure`.

Comment (KipHamiltons) [reply src/book/03-guides/02-tools/03-webots.mdx:102]: I might make a `details` with those details

Comment (KipHamiltons) [reply src/book/03-guides/02-tools/03-webots.mdx:111]: I think that's a lot nicer

**Review by CMurtagh-LGTM — DISMISSED** [review]
LGTM

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 2]
```diff
@@ -32,7 +34,9 @@ slug: /guides/general/glossary
 
 [**Clang**](https://clang.llvm.org/): A compiler, frontend, and tooling infrastructure used for C and C++ which is part of the LLVM infrastructure.
 
-[**ClangFormat**](https://clang.llvm.org/docs/ClangFormat.html): A software which formats source code according to a given set of rules.
+[**ClangFormat**](https://clang.llvm.org/docs/ClangFormat.html): A package which formats source code according to a given set of rules.
+
+[**ClangTidy**](https://clang.llvm.org/extra/clang-tidy/): A package which statically analyses source code, generating warnings based on arbitrary defined rules you provide.
```
Comment (Bidski) [inline src/book/03-guides/04-general/03-glossary.mdx:39]:
```suggestion
[**ClangTidy**](https://clang.llvm.org/extra/clang-tidy/): A package which statically analyses source code, generating warnings based on arbitrarily defined rules you provide.
```

File: src/book/03-guides/02-tools/03-webots.mdx [hunk 3]
```diff
@@ -71,7 +71,7 @@ Install libraries for cmake, protobuf, eigen, and yaml-cpp:
 
 ```sh
 sudo apt-get update
-sudo apt-get install cmake-curses-gui libprotobuf-dev protobuf-compiler libeigen3-dev libyaml-cpp-dev
+sudo apt-get install cmake-curses-gui libprotobuf-dev protobuf-compiler libeigen3-dev libyaml-cpp-dev ninja-build
```
Comment (Bidski) [inline src/book/03-guides/02-tools/03-webots.mdx:74]: The instructional text above this block needs editing

File: src/book/03-guides/02-tools/03-webots.mdx [hunk 4]
```diff
@@ -96,25 +96,38 @@ sudo apt-get install cmake-curses-gui libprotobuf-dev protobuf-compiler libeigen
...
+   | --------------------------------- | -------------------------------------------------------------------------------------------------------------- | ------------------- |
+   | `-DUSE_ASAN=ON`                   | Compile with [address sanitizer](https://clang.llvm.org/docs/AddressSanitizer.html)                            | `OFF`               |
+   | `-DUSE_UBSAN=ON`                  | Compile with [undefined behaviour sanitizer](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html)      | `OFF`               |
+   | `-DENABLE_CLANG_TIDY=ON`          | Analyse the code being compiled with [clang-tidy](https://clang.llvm.org/extra/clang-tidy/)                    | `OFF`               |
+   | `-DWEBOTS_HOME=path/to/webots`    | Tells the compiler where webots is (usually implicitly set by environment variable)                            | `/usr/local/webots` |
+   | `-DCMAKE_BUILD_TYPE=<build type>` | Tells the compiler what type of build type to use, out of {`Release`, `Debug`, `MinSizeRel`, `RelWithDebInfo`} | `Release`           |
```
Comment (Bidski) [inline src/book/03-guides/02-tools/03-webots.mdx:122]: These are not the **only** flags that can be set. Almost everything that appears in `ccmake` could be specified as a flag on the command line. Maybe reword it to something like this

```
where `-i` runs an interactive configuration with `ccmake`, and `<flags>` can be any valid cmake command line flag. Some common cmake flags are:
```

**Review by Bidski — CHANGES_REQUESTED** [review]

File: src/book/03-guides/02-tools/03-webots.mdx (hunk 4 above)
Comment (KipHamiltons) [reply src/book/03-guides/02-tools/03-webots.mdx:122]: It's a good point. I have said `Some useful cmake flags are` instead though, since these aren't common cmake flags per se, they're just the ones you might commonly use for this repo.

**Review by Bidski — APPROVED** [review]
LGTM
